tracker/tracker-assist/src/ScreenRecordingState.ts

```diff
@@ -0,0 +1,121 @@
+import ConfirmWindow from './ConfirmWindow/ConfirmWindow.js'
+import { recordRequestDefault, } from './ConfirmWindow/defaults.js'
+import type { Options as AssistOptions, } from './Assist'
+
+export enum RecordingState {
+  Off,
+  Requested,
+  Recording,
+}
+
+const borderStyles = {
+    height: '100vh',
+    width: '100vw',
+  border: '2px dashed red',
+  left: 0,
+  top: 0,
+  position: 'fixed',
+  'pointer-events': 'none',
+}
+
+const buttonStyles = {
+  cursor: 'pointer',
+  color: 'white',
+  position: 'fixed',
+  bottom: '0',
+  left: 'calc(50vw - 60px)',
+  'font-weight': 500,
+  padding: '2px 4px',
+  background: '#394EFF',
+  'border-top-right-radius': '3px',
+  'border-top-left-radius': '3px',
+  'text-align': 'center',
+}
+
+export default class ScreenRecordingState {
+  private status = RecordingState.Off
+  private agentsRecordingSession: string[] = []
+  private overlayAdded = false
+
+  constructor(
+    private readonly onAccept: () => void,
+    private readonly onDeny: () => void,
+    private readonly options: AssistOptions
+  ) {}
+
+  public get isActive() {
+    return this.status !== RecordingState.Off
+  }
+
+  private confirm: ConfirmWindow | null = null
+
+  public requestRecording = (id: string) => {
+    if (this.status !== RecordingState.Off) return
+    this.status = RecordingState.Requested
+
+    this.confirm = new ConfirmWindow(recordRequestDefault(this.options.controlConfirm))
+    this.confirm.mount().then(allowed => {
+      if (allowed) {
+        this.acceptRecording()
+        this.agentsRecordingSession.push(id)
+      } else {
+        this.confirm?.remove()
+        this.rejectRecording()
+      }
+    })
+    .then(() => {
+      this.confirm?.remove()
+    })
+    .catch(e => {
+        this.confirm?.remove()
+        console.error(e)
+      })
+  }
+
+  private readonly acceptRecording = () => {
+    if (!this.overlayAdded) {
+      const stopButton = window.document.createElement('div')
+      stopButton.onclick = () => this.rejectRecording()
+      Object.assign(stopButton.style, buttonStyles)
+      stopButton.textContent = 'Stop Recording'
+      stopButton.id = 'or-recording-button'
+      stopButton.setAttribute('data-openreplay-obscured', '')
+      stopButton.setAttribute('data-openreplay-hidden', '')
+      stopButton.setAttribute('data-openreplay-ignore', '')
+      window.document.body.appendChild(stopButton)
+
+      const borderWindow = window.document.createElement('div')
+      Object.assign(borderWindow.style, borderStyles)
+      borderWindow.id = 'or-recording-border'
```

Comment:
why don't you want to use class property instead like in every other component? 
This would be more reliable.

Reply:
m?